src/PrMerger.js

```diff
@@ -5,127 +5,146 @@ const Logger = Log.Logger;
-            assert(!result.delayed());
-            if (result.succeeded() || result.failed())
-                return true;
-            // This result is when one of'dry run' bot options is on.
-            // Will wait while this option is on the way.
-            assert(result.suspended());
-        } else {
-            assert(processor.isInitiator());
-            if (result.failed() || result.suspended())
-                return true;
-
-            if (result.delayed()) {
-                if (this.rerunIn === null || this.rerunIn > result.delay())
-                    this.rerunIn = result.delay();
-                assert(this.rerunIn);
-                return true;
+        for (let tag of rawTags) {
+            let prNum = Util.ParseTag(tag.ref);
+            if (prNum === null)
+                continue;
+            for (let pr of this._todo) {
+                if (prNum === pr.number.toString()) {
+                    prNum = null;
+                    this._tags.push(tag);
+                    break;
+                }
+            }
+            if (prNum !== null) {
+                if (!Config.dryRun())
+                    await GH.deleteReference(Util.StagingTag(prNum));
             }
-
-            assert(result.succeeded());
         }
-        return false;
     }
 
-    // Loads 'being-in-merge' PR, if exists (the PR has tag and staging_branch points to the tag).
+    // Returns a raw PR with a staged commit (or null).
+    // If that PR exists, it is in either a "staged" or "merged" state.
+    // Requires _importTags() being run first.
     async _current() {
         Logger.info("Looking for the current PR...");
         const stagingSha = await GH.getReference(Config.stagingBranchPath());
-        // request all repository tags
-        let tags = await GH.getTags();
         // search for a tag, the staging_branch points to,
         // and parse out PR number from the tag name
-        const tag = tags.find((t) => { return (t.object.sha === stagingSha) && Util.MatchTag(t.ref); });
+        const tag = this._tags.find((t) => { return (t.object.sha === stagingSha) && Util.MatchTag(t.ref); });
         if (tag === undefined) {
             Logger.info("No current PR found.");
             return null;
         }
         const prNum = Util.ParseTag(tag.ref);
         Logger.info("PR" + prNum + " is the current");
-        const stagingPr = await GH.getPR(prNum, false);
-        return new MergeFinalizer(stagingPr, stagingSha);
-    }
-
-    logStatistics() {
-        Logger.info("Merge step finished. Total PRs processed: " + this.total + ", skipped due to errors: " + this.errors);
+        const pr = GH.getPR(prNum, false);
+        if (pr.state === 'open')
+            return pr;
+        Logger.warn("The current PR" + prNum + " unexpectedly closed.");
+        return null;
     }
 } // PrMerger
 
-module.exports = PrMerger;
+// PrMerger's create and execute() wrapper.
+module.exports = async function Step() {
```

Comment:
If possible, please split into two statements. The Step() function definition is complex enough on its own to keep it isolated from the exporting concerns/statement.

Reply:
Done.